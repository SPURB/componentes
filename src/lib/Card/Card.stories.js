import { storiesOf } from '@storybook/vue'
import Card from './Card.vue'

storiesOf('Card', module).add('Card', () => ({
  components: { Card },
  template: `
      <Card headerClass="bg-gray-200 px-2 py-3">
        <template v-slot:header>
          <h1>Aqui pode estar o título do card</h1>
        </template>
        <template v-slot:body>
          <p>Um parágrafo para o conteúdo principal.</p>
          <p>E um outro parágrafo.</p>          
        </template>
        <template v-slot:footer>
          <p>Aqui está o footer</p>
        </template>
      </Card>
    
    `
}))
