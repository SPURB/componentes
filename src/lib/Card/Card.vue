<template>
  <div class="max-w-sm rounded overflow-hidden shadow-lg">
    <div :class="headerClass">
      <slot name="header" />
    </div>
    <div :class="bodyClass">
      <slot name="body" />
    </div>
    <div :class="footerClass">
      <slot name="footer" />
    </div>
  </div>
</template>

<script>
export default {
  name: 'Card',
  props: {
    headerClass: {
      type: String,
      default: 'px-6 py-4'
    },
    bodyClass: {
      type: String,
      default: 'px-6 py-4'
    },
    footerClass: {
      type: String,
      default: 'px-6 pt-4 pb-2'
    }
  }
}
</script>

<style>
</style>