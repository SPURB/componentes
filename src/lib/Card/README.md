# Combobox.vue
Componente de dropdown para seleção de items

## Exemplo
https://spurb.github.io/componentes/?path=/story/combobox--normal

## Instruções de uso
Declare
```js
import { Card } from '@spurb/componentes'
export default {
  components: { Card }
}
```

Exemplo de uso no template
```html
<Card headerClass="bg-gray-200 px-2 py-3">
  <template v-slot:header>
    <h1>Aqui pode estar o título do card</h1>
  </template>
  <template v-slot:body>
    <p>Um parágrafo para o conteúdo principal.</p>
    <p>E um outro parágrafo.</p>          
  </template>
  <template v-slot:footer>
    <p>Aqui está o footer</p>
  </template>
</Card>
```

### Props
| nome | data type | descrição |
|:-|:-|:-|
| `headerClass` |  String | Define as classes a serem usadas no header do card |
| `bodyClass` | String | Define as classes a serem usadas no body do card |
| `footerClass` | String | Define as classes a serem usadas no footer do card |

### Slots
| nome | descrição
| `header` | Define conteúdo no header do card |
| `body` | Define conteúdo no body do card |
| `footerClass` | Define conteúdo no footer do card|
