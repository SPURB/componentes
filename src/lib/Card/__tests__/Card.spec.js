import { shallowMount } from '@vue/test-utils'
import Card from '../Card.vue'

const headerClass = 'bg-gray-200 px-2 py-3'
const bodyClass = 'bg-gray-400 px-2 py-3'
const footerClass= 'bg-gray-200 px-2 py-3'

describe('Card.vue', () => {
  it('Renderiza corretamente passando props', () => {
    const wrapper = shallowMount(Card, {
      propsData: { headerClass, bodyClass, footerClass },
      slots: {
        header: '<h1>Aqui pode estar o título do card</h1>',
        body: '<p>Um parágrafo para o conteúdo principal.</p><p>E um outro parágrafo.</p>',
        footer: '<p>Aqui está o footer</p>'
      }
    })
    expect(wrapper.element).toMatchSnapshot() // compara renderização por snashot
  })
})
