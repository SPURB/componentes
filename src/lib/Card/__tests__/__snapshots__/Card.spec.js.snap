// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.vue Renderiza corretamente passando props 1`] = `
<div
  class="max-w-sm rounded overflow-hidden shadow-lg"
>
  <div
    class="bg-gray-200 px-2 py-3"
  >
    <h1>
      Aqui pode estar o título do card
    </h1>
  </div>
   
  <div
    class="bg-gray-400 px-2 py-3"
  >
    <p>
      Um parágrafo para o conteúdo principal.
    </p>
    <p>
      E um outro parágrafo.
    </p>
  </div>
   
  <div
    class="bg-gray-200 px-2 py-3"
  >
    <p>
      Aqui está o footer
    </p>
  </div>
</div>
`;
